### PR TITLE
osdocs771 AWS private clusters

### PR DESCRIPTION
**Pull request by kalexand-rh**
https://jira.coreos.com/browse/OSDOCS-771

@abhinavdahiya, I'm working on the modules for installing private clusters. Will you PTAL at the new content and let me know what else is required? I'm going to make the BYO VPC steps a requirement for this story, but is there anything that I'm missing?

### CODE REVIEW

File: modules/private-clusters-about-aws.adoc [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// Module included in the following assemblies:
+//
+// * installing/
+
+[id="private-clusters-about-aws_{context}"]
+= Private clusters in AWS
+
+To create a private cluster on Amazon Web Services (AWS), you must provide an existing private VPC and subnets to host the cluster. The installation program must also be able to resolve the DNS records that the cluster requires. The installation program configures the Ingress Operator and API server for only internal traffic.
```
Comment (abhinavdahiya) [inline modules/private-clusters-about-aws.adoc]: > only internal traffic

Maybe replace it with something closer to, ` for access from the private network only`

**Comment by abhinavdahiya** [comment]
lgtm

**Comment by kalexand-rh** [comment]
@jianlinliu, will you PTAL? I cherry-picked the existing VPC change onto this PR so that I could build the assembly, so many of the changes are duplicated from that PR. 

The content that's new for this PR is:
installing/installing_aws/installing-aws-private.adoc
modules/private-clusters-default.adoc 
modules/private-clusters-about-aws.adoc 
modules/installation-aws-config-yaml.adoc, line 102 and its note
modules/installation-configuration-parameters.adoc, line 138

**Comment by jianlinliu** [comment]
Hmm, personally I think using existing VPC and private cluster are two different scenarios. Of course, user can combine the two scenarios for different purpose. E.g: if user want a private cluster, user even can run openshift-install binary in their laptop, but eventually installation will report a fake negative result to fail the whole installation, because installer can not access the cluster api and *.apps endpoint outside the vpc, but actually the cluster is already provisioned successfully. To make the whole progress more friendly, maybe we could suggest user firstly create a VPC, create a bastion host in this VPC, download installer binary onto this bastion host, run the whole installation on this bastion host to create a private cluster. From my understanding, an existing VPC is not a MUST from technical perspective for a private cluster. I also go through the whole preview, seem like we did not emphasise why we must have an existing VPC, did not mention where installation should be happening either (should be in a bastion host). 
For existing VPC, some customers maybe indeed want create multiple clusters facing internet in one shared VPC, nothing with private cluster.
So here we combine the two different scenarios into one topic, probably confuse customers.

**Comment by kalexand-rh** [comment]
@jianlinliu, the enhancement makes it sound like the VPC is required: https://github.com/openshift/enhancements/blob/master/enhancements/installer/aws-internal-clusters.md#proposal

**Comment by jianlinliu** [comment]
Yeah, from customer use experience, to be more friendly, pre-existing VPC can be treated as a requirement.
My last comment is suggesting we still need a separate section for BYO VPC. I thought this PR will deprecate https://github.com/openshift/openshift-docs/pull/18729, seem like not. If that, I am okay with an existing VPC is required.
If that, the only comments is:

> seem like we did not emphasise why we must have an existing VPC, did not mention where installation should be happening either (should be in a bastion host).

File: modules/private-clusters-default.adoc [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_aws/installing-aws-private.adoc
+// * installing/installing_gcp/installing-gcp-private.adoc
+
+[id="private-clusters-default_{context}"]
+= Private clusters
+
+If your environment does not require an external internet connection, you can deploy a private {product-title} cluster that does not expose external endpoints. Private clusters are accessible from only an internal network and are not visible to the Internet.
```
Comment (kalexand-rh) [inline modules/private-clusters-default.adoc]: @jianlinliu, does this module address your concerns about context for why you need your own network and using the bastion host?

**Comment by kalexand-rh** [comment]
@openshift/team-documentation, PTAL?

**Review by sheriff-rh — APPROVED** [review]
Excellent work! All changes look good to me.

File: modules/private-clusters-default.adoc (hunk 2 above)
Comment (jianlinliu) [reply modules/private-clusters-default.adoc]: LGTM.

**Pull request merged**